src/spyglass/position/v1/imported_pose.py

```diff
@@ -0,0 +1,152 @@
+                    timestamps, filename=nwb_file_name
+                )
+                valid_intervals = get_valid_intervals(
+                    timestamps,
+                    sampling_rate=sampling_rate,
+                    min_valid_len=sampling_rate,
+                )
+                interval_key = {
+                    "nwb_file_name": nwb_file_name,
+                    "interval_list_name": f"pose_{name}_valid_intervals",
+                    "valid_times": valid_intervals,
+                    "pipeline": "ImportedPose",
+                }
+                interval_keys.append(interval_key)
+
+                # master key
+                master_key = {
+                    "nwb_file_name": nwb_file_name,
+                    "interval_list_name": interval_key["interval_list_name"],
+                    "pose_object_id": obj.object_id,
+                    "skeleton_object_id": obj.skeleton.object_id,
+                }
+                master_keys.append(master_key)
+
+                # part keys
+                for part, part_obj in obj.pose_estimation_series.items():
+                    part_key = {
+                        "nwb_file_name": nwb_file_name,
+                        "interval_list_name": interval_key[
+                            "interval_list_name"
+                        ],
+                        "part_name": part,
+                        "part_object_id": part_obj.object_id,
+                    }
+                    part_keys.append(part_key)
+
+        IntervalList().insert(interval_keys, skip_duplicates=True)
+        self.insert(master_keys, skip_duplicates=True)
+        self.BodyPart().insert(part_keys, skip_duplicates=True)
+
+    def fetch_pose_dataframe(self, key=dict()):
+        """Fetch pose data as a pandas DataFrame
+
+        Parameters
+        ----------
+        key : dict
+            Key to fetch pose data for
+
+        Returns
+        -------
+        pd.DataFrame
+            DataFrame containing pose data
+        """
+        key = (self & key).fetch1("KEY")
+        pose_estimations = (
+            (self & key).fetch_nwb()[0]["pose"].pose_estimation_series
+        )
+
+        index = None
+        pose_df = {}
+        for body_part in pose_estimations.keys():
+            if index is None:
+                index = pd.Index(
+                    pose_estimations[body_part].timestamps[:],
+                    name="time",
+                )
+
+            part_df = {
+                "video_frame_ind": np.nan,
+                "x": pose_estimations[body_part].data[:, 0],
+                "y": pose_estimations[body_part].data[:, 1],
+                "likelihood": pose_estimations[body_part].confidence[:],
+            }
+
+            pose_df[body_part] = pd.DataFrame(part_df, index=index)
+
+        pose_df
+        return pd.concat(pose_df, axis=1)
+
+    def fetch_skeleton(self, key=dict()):
+        nwb = (self & key).fetch_nwb()[0]
```

Comment:
What would happen if this func was passed an under-specified key? It looks like this might take the first nwb of multiple and ignore the rest. That case might be worth an error or logging a warning and returning

```
query = self & key
if len(query) != 1:
    raise ValueError(f"Key should specify one entry. Found {len(query)}")
```